minio/sse.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# Minio Python Library for Amazon S3 Compatible Cloud Storage, (C) 2015 Minio, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+minio.sse
+~~~~~~~~~~~~~~~~~~~
+
+This module contains core API parsers.
+
+:copyright: (c) 2015 by Minio, Inc.
+:license: Apache 2.0, see LICENSE for more details.
+
+"""
+import base64
+import hashlib
+
+class SSEC(object):
+
+    def __init__(self, key):
+        self.key = key
+
+    def type(self):
+        return "SSE-C"
+    
+    def marshal(self):
+
+        b64key = base64.b64encode(self.key)
+        md5 = hashlib.md5()
+        md5.update(self.key)
+        md5_key = base64.b64encode(md5.digest()).decode()
+        keys = {
+            "X-Amz-Server-Side-Encryption-Customer-Algorithm": "AES256",
+            "X-Amz-Server-Side-Encryption-Customer-Key": b64key.decode(),
+            "X-Amz-Server-Side-Encryption-Customer-Key-MD5": md5_key
+            }
+
+        return keys
+
+class copy_SSEC(object):
+
+    def __init__(self, key):
+        self.key = key
```

Comment:
if key is not 32 bytes, need to throw error - SSE-C keys need to be 256 bit base64 encoded.